compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1205,6 +1205,14 @@ fn compare_number_of_generics<'tcx>(
         return Ok(());
     }
 
+    // Skip emitting errors for RPITITs, we've already checked generics for trait's and impl's
+    // functions.
+    if tcx.opt_rpitit_info(trait_.def_id).is_some() {
+        return Err(tcx
+            .sess
+            .delay_span_bug(rustc_span::DUMMY_SP, "error should have been emitted"));
```

Comment:
this could use a better error message, since devs will see it with `-Ztreat-err-as-bug`.